app/src/sharedTest/java/org/oppia/app/parser/FractionInteractionViewModelTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.UserAnswer
+
+@RunWith(AndroidJUnit4::class)
+class StringToNumberWithUnitsParserTest {
+
+  private var context: Context? = null
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    ApplicationProvider.getApplicationContext<Context>().resources.configuration.orientation =
+      Configuration.ORIENTATION_LANDSCAPE
+    context = ApplicationProvider.getApplicationContext()
+  }
+
+  fun getPendingAnswer(answerText: String): UserAnswer {
+    val userAnswerBuilder = UserAnswer.newBuilder()
+    if (answerText.isNotEmpty()) {
+      val answerTexttring = answerText.toString()
+      userAnswerBuilder.answer = InteractionObject.newBuilder()
+        .setNumberWithUnits(StringToNumberWithUnitsParser().parseNumberWithUnits(answerTexttring))
+        .build()
+      userAnswerBuilder.plainAnswer = answerTexttring
+    }
+    return userAnswerBuilder.build()
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withNoText_hasCorrectPendingAnswerType() {
+
+    val interactionObject = getPendingAnswer("").answer
+    assertThat(interactionObject).isInstanceOf(InteractionObject::class.java)
+    assertThat(getPendingAnswer("").answer.real).isWithin(1e-5).of(0.0)
+    assertThat(interactionObject.numberWithUnits.fraction.denominator).isEqualTo(0)
+    assertThat(interactionObject.numberWithUnits.fraction.numerator).isEqualTo(0)
+    assertThat(interactionObject.numberWithUnits.fraction.wholeNumber).isEqualTo(0)
+    assertThat(interactionObject.numberWithUnits.fraction.isNegative).isEqualTo(false)
+    assertThat(interactionObject.numberWithUnits.unitOrBuilderList).isEmpty()
+
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withNegativeWholeNumberText_hasCorrectPendingAnswer() {
+    val interactionObject = getPendingAnswer("-9").answer
+    assertThat(interactionObject).isInstanceOf(InteractionObject::class.java)
+    assertThat(interactionObject.objectTypeCase).isEqualTo(InteractionObject.ObjectTypeCase.NUMBER_WITH_UNITS)
+    assertThat(interactionObject.numberWithUnits.real).isLessThan(0.0)
+    assertThat(interactionObject.numberWithUnits.real).isEqualTo(-9.0)
```

Comment:
Here & everywhere else: real values need to be compared using isWithin() since exact float equality shouldn't be relied upon when the float values are being computed.